src/components/App/App.js

```diff
@@ -0,0 +1,362 @@
+        //setToken(res.token);
+        history.push("/");
+      } 
+    })
+    .catch((err) => {
+      console.log(err);
+      if (err === 'Error 400')
+      setIsClientError('Wrong Email or Password ');
+      else{
+        setIsClientError('Something else wrong on handleLogin function ');
+      }
+    })
+    .finally(() => {
+      setIsCheckingToken(false);
+    });
+}
+
+function signout() {
+  setIsLoggedIn(false);
+  setCurrentUser({});
+  localStorage.removeItem('jwt');
+  setIsLoginPopupOpen(true);
+}
+
+
+//article handlers
+
+//search
+
+function handleSearch({searchTerm}){
+if (searchTerm !== ''){
+  setIsLoading(true);
+  setKeyword(searchTerm);
+  newsApi.getArticles(searchTerm)
+    .then((res) => {
+      console.log('getArticles on handleSearch', searchTerm)
+      if (res.articles) {
+        setSearchData({searchTerm})
+        articles.current = res.articles;
+        setFoundArticles(articles.current.slice(0, 3));
+        console.log(articles.current.slice(0,1));
+        console.log(articles.source);
+      }
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      setIsLoading(false);
+    })
+}
+}
+
+
+//save
+
+function handleSave({keyword, title, description, source, publishedAt, url, urlToImage }) {
+  mainApi.saveArticle({keyword: source.name, source: source.name, title, text: description, date: publishedAt, link: url, image: urlToImage}) //do I need token there?
+  .then((savedArticle) => {
+    setSavedArticles([savedArticle, ...savedArticles])
+  })
+  .catch((err) => {
+    console.log(err);
+  })
+}
+
+//remove from saved
+
+function handleRemove(cardId) {
+  savedArticles.map((removedCard) => {
+      cardId = removedCard._id;
+      console.log(cardId);
+    
+  })
+  mainApi.deleteArticle(cardId)
+      .then((res) => {
+        setCurrentUser((currentUser) => ({
+          ...currentUser, savedArticles: savedArticles.map((removedCard) => removedCard._id !== res._id) 
+        }))
+    })
+    .catch((err) => console.log(err))
+}
+
+//show more cards
+
+function handleMore(){
+  setFoundArticles(articles.current.slice(0, foundArticles.length + 3));
+  console.log('handle more ' + (articles.current.slice(0, foundArticles.length + 3)))
+}
+
+
+
+//popups
+
+function closeAllPopups() {
+  setIsLoginPopupOpen(false);
+  setIsRegisterPopupOpen(false);
+  setIsInfoToolTipOpen(false);
+  setIsMobileMenuOpen(false);
+}
+
+  function handleLoginClick() {
+      setIsLoginPopupOpen(true);
+      setIsClientError('');
+    }
+  
+    function handleRegisterClick() {
+      setIsClientError('');
+      setIsRegisterPopupOpen(true);
+    }
+
+    function handleMobileMenuClick(){
+      setIsMobileMenuOpen(true);
+    }
+
+    function handleSearchClick(){
+      setIsLoading(true);
+    }
+
+
+    return(
+        <CurrentUserContext.Provider value={currentUser}>
+        <div className="app">
+        <MobileMenu
+        isOpen={isMobileMenuOpen}
```

Comment:
You need to format the markup to show indentation and to make the code more readable. You can do that with `prettier` or a combination `SHIFT + ALT + F`


```jsx


  return (
    <CurrentUserContext.Provider value={currentUser}>
      <div className="app">
        <MobileMenu
          isOpen={isMobileMenuOpen}

```